src/player.cpp

```diff
@@ -5484,15 +5483,17 @@ void player::suffer()
     const bool leafier = has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
     const bool leafiest = has_trait( trait_LEAVES3 );
     int sunlight_nutrition = 0;
-    if( leafy && g->is_in_sunlight( pos() ) ) {
+    if( leafy && g->m.is_outside( pos() ) && g->light_level( pos().z ) ) {
```

Comment:
Finally got around to testing this, only to find this returns 1 at *night*, meaning it makes plant mutants photosynthesize any time they are outdoors regardless of light level.